dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dashboard/DashboardExportSearchResultsPageTest.java

```diff
@@ -55,28 +51,29 @@ public void setUp() {
         this.type2.setId(2L);
         this.type2.setTitle("abc");
 
-        when(this.datasetFiledTypeRepo.findAll())
-                .thenReturn(asList(this.type1, this.type2));
+        when(this.datasetFiledTypeRepo.findAll()).thenReturn(types);
+
+        this.page = new DashboardExportSearchResultsPage(this.session, null,
```

Comment:
Why did you remove `@InjectMocks`?

Reply:
the constructor  of DashboardExportSearchResultsPage invokes datasetFiledTypeRepo.findAll().

i need to call when(this.datasetFiledTypeRepo.findAll()).thenReturn(types); BEFORE instantiating DashboardExportSearchResultsPage 
otherwise datasetFiledTypeRepo.findAll() returns empty list

Reply:
I guess the problem stems from using logic inside the constructor,  which isn't always the best choice (specifically in a dependency injection context). For instance, you could have extracted the logic to a `@PostCostruct` annotated method and call that method here to make the intend (logic usage) more obvious.

Reply:
fixed